shared/js/smart-screen/f_t_e_wizard.js

```diff
@@ -0,0 +1,192 @@
+    this._pages.forEach((page, idx) => this._hide(idx));
+    this._show(this._currentPage);
+    this._updateNavigation();
+
+    this._running = true;
+  };
+
+  FTEWizard.prototype.uninit = function fw_uninit() {
+    if (this._launched) {
+      return;
+    }
+    this._container.removeEventListener('keydown', this);
+    this._container.removeEventListener('keyup', this);
+    this._simpleKeyNavigation.stop();
+
+    this._running = false;
+  };
+
+  FTEWizard.prototype.goNext = function fw_next() {
+    if (this._currentPage >= this._pages.length - 1) {
+      this.finish();
+      return;
+    }
+
+    this._hide(this._currentPage);
+    this._currentPage++;
+    this._show(this._currentPage);
+    this._updateNavigation();
+  };
+
+  FTEWizard.prototype.goPrev = function fw_prev() {
+    if (this._currentPage <= 0) {
+      return;
+    }
+
+    this._hide(this._currentPage);
+    this._currentPage++;
+    this._show(this._currentPage);
+    this._updateNavigation();
+  };
+
+  FTEWizard.prototype.finish = function fw_finish() {
+    this._container.classList.add('finish');
+
+    attachTransitionEnd(this._container, () => {
+
+      this._hide(this._currentPage);
+      this._container.style.display = 'none';
+      this.uninit();
+      this._finished = true;
+      this._launched = true;
+
+      localStorage.setItem(this._name + '.fteskip', true);
+      this._onfinish && this._onfinish();
+    });
+  };
+
+  FTEWizard.prototype.focus = function fw_focus() {
+    this._simpleKeyNavigation.focus();
+  };
+
+  FTEWizard.prototype._updateNavigation = function fw_updateNavigation() {
+    var list = Array.from(this._pages[this._currentPage].
+                                    getElementsByClassName(this._buttonsClass));
+
+    this._simpleKeyNavigation.updateList(list);
+  };
+
+  FTEWizard.prototype._hide = function fw_hide(pageIdx) {
+    this._pages[pageIdx].style.display = 'none';
+  };
+
+  FTEWizard.prototype._show = function fw_hide(pageIdx) {
+    this._pages[pageIdx].style.display = '';
+  };
+
+  FTEWizard.prototype.handleEvent = function fw_handleEvent(evt) {
+    switch(evt.type) {
+
+      case 'keyup':
+        if (evt.keyCode === KeyEvent.DOM_VK_RETURN) {
+          switch (evt.target.dataset.behavior) {
+
+            case 'next':
+              this.goNext();
+              break;
+
+            case 'prev':
+              this.goPrev();
+              break;
+
+            case 'finish':
+              this.finish();
+              break;
+          }
+        }
+        // By default we stop propagating keyevent such that the main logic of
+        // app is not triggered while FTE is working.
+        !this.propagateKeyEvent && evt.stopPropagation();
+        break;
+      case 'keydown':
+        // also stop propagating keydown since arrow keys are triggered by
+        // keydown in smart-screen apps.
+        !this.propagateKeyEvent && evt.stopPropagation();
+        break;
+    }
```

Comment:
I think `keypress` is plenty since mouse events should follow the bubbling/capturing already.